databricks/koalas/indexes.py

```diff
@@ -1631,6 +1632,67 @@ def repeat(self, repeats: int) -> "Index":
         else:
             return ks.concat([kdf] * repeats).index
 
+    def union(self, other, sort=None):
+        """
+        Form the union of two Index objects.
+
+        Parameters
+        ----------
+        other : Index or array-like
+        sort : bool or None, default None
+            Whether to sort the resulting Index.
+
+        Returns
+        -------
+        union : Index
+
+        Examples
+        --------
+
+        Index
+
+        >>> idx1 = ks.Index([1, 2, 3, 4])
+        >>> idx2 = ks.Index([3, 4, 5, 6])
+        >>> idx1.union(idx2).sort_values()
+        Int64Index([1, 2, 3, 4, 5, 6], dtype='int64')
+
+        MultiIndex
+
+        >>> midx1 = ks.MultiIndex.from_tuples([("x", "a"), ("x", "b"), ("x", "c"), ("x", "d")])
+        >>> midx2 = ks.MultiIndex.from_tuples([("x", "c"), ("x", "d"), ("x", "e"), ("x", "f")])
+        >>> midx1.union(midx2).sort_values()  # doctest: +SKIP
+        MultiIndex([('x', 'a'),
+                    ('x', 'b'),
+                    ('x', 'c'),
+                    ('x', 'd'),
+                    ('x', 'e'),
+                    ('x', 'f')],
+                   )
+        """
+        sort = True if sort is None else sort
+        sort = validate_bool_kwarg(sort, "sort")
+        if not isinstance(other, Index) and isinstance(self, MultiIndex):
+            if isinstance(other, list) and not all([isinstance(item, tuple) for item in other]):
+                raise TypeError("other must be a MultiIndex or a list of tuples")
+            other = MultiIndex.from_tuples(other)
+        if not isinstance(other, Index) and not isinstance(self, MultiIndex):
+            other = Index(other)
+        if type(self) is not type(other):
+            # TODO: We can't support different type of values in a single column for now.
+            raise NotImplementedError("Union between Index and MultiIndex is not yet supported")
+        sdf_self = self._internal._sdf.select(self._internal.index_spark_columns)
+        sdf_other = other._internal._sdf.select(
+            other._internal.index_spark_columns
+        ).drop_duplicates()
+        sdf = sdf_self.union(sdf_other.subtract(sdf_self))
+        if isinstance(self, MultiIndex):
+            sdf = sdf.drop_duplicates()
```

Comment:
Why only for `MultiIndex`?

Reply:
pandas has different behavior between `Index` and `MultiIndex`.

For `Index`, they use all of `self` even they are duplicated.

```python
>>> idx1 = pd.Index([1, 2, 3, 4, 3, 4, 3, 4])
>>> idx2 = pd.Index([3, 4, 5, 6])
>>> idx1.union(idx2)
Int64Index([1, 2, 3, 3, 3, 4, 4, 4, 5, 6], dtype='int64')
```

But for `MultiIndex`, they get rid of the duplicated values from `self`.

```python
>>> pmidx1 = pd.MultiIndex.from_tuples([(1, 1), (1, 2), (1, 3), (1, 4), (1, 3), (1, 4)])
>>> pmidx2 = pd.MultiIndex.from_tuples([(1, 3), (1, 4), (1, 5), (1, 6)])
>>> pmidx1.union(pmidx2)
MultiIndex([(1, 1),
            (1, 2),
            (1, 3),
            (1, 4),
            (1, 5),
            (1, 6)],
           )
```

That is the reason i just use `drop_duplicates()` for `MultiIndex` only

Reply:
I see. Shall we add the example in the test?

Reply:
@ueshin Sure, I'll add.

(Anyway, sometimes It is really hard to understand of pandas' behavior 😂)